jsonrpc/txpool_endpoint_test.go

```diff
@@ -152,6 +152,127 @@ func TestContentEndpoint(t *testing.T) {
+		assert.Equal(t, testTx2.Gas(), uint64(txData.Gas))
+		assert.Nil(t, testTx2.GasPrice())
+		assert.Equal(t, *(testTx2.GasFeeCap()), big.Int(*txData.GasFeeCap))
+		assert.Equal(t, *(testTx2.GasTipCap()), big.Int(*txData.GasTipCap))
+		assert.Equal(t, testTx2.To(), txData.To)
+		assert.Equal(t, testTx2.From(), txData.From)
+		assert.Equal(t, *(testTx2.ChainID()), big.Int(*txData.ChainID))
+		assert.Equal(t, *(testTx2.Value()), big.Int(txData.Value))
+		assert.Equal(t, testTx2.Input(), []byte(txData.Input))
+		assert.Equal(t, (*argUint64)(nil), txData.BlockNumber)
+		assert.Equal(t, (*argUint64)(nil), txData.TxIndex)
+	})
+
+	t.Run("returns correct data for queued transaction", func(t *testing.T) {
+		t.Parallel()
+
+		mockStore := newMockTxPoolStore()
+		address1, address2 := types.Address{0x1}, types.Address{0x2}
+		testTx1 := newTestTransaction(2, address1)
+		testTx2 := newTestDynamicFeeTransaction(1, address2)
+		mockStore.queued[address1] = []*types.Transaction{testTx1}
+		mockStore.queued[address2] = []*types.Transaction{testTx2}
+		txPoolEndpoint := &TxPool{mockStore}
+
+		result, _ := txPoolEndpoint.ContentFrom(address2)
+
+		response := result.(ContentAddressResponse)
+		assert.Equal(t, 1, len(response.Queued))
+
+		txData := response.Queued[testTx2.Nonce()]
+		assert.NotNil(t, txData)
+		assert.Equal(t, (argUint64)(types.DynamicFeeTxType), txData.Type)
+		assert.Equal(t, testTx2.Gas(), uint64(txData.Gas))
+		assert.Nil(t, testTx2.GasPrice())
+		assert.Equal(t, *(testTx2.GasFeeCap()), big.Int(*txData.GasFeeCap))
+		assert.Equal(t, *(testTx2.GasTipCap()), big.Int(*txData.GasTipCap))
+		assert.Equal(t, testTx2.To(), txData.To)
+		assert.Equal(t, testTx2.From(), txData.From)
+		assert.Equal(t, *(testTx2.ChainID()), big.Int(*txData.ChainID))
+		assert.Equal(t, *(testTx2.Value()), big.Int(txData.Value))
+		assert.Equal(t, testTx2.Input(), []byte(txData.Input))
+		assert.Equal(t, (*argUint64)(nil), txData.BlockNumber)
+		assert.Equal(t, (*argUint64)(nil), txData.TxIndex)
+	})
+
+	t.Run("returns correct ContentAddressResponse data for multiple transactions", func(t *testing.T) {
+		t.Parallel()
+
+		mockStore := newMockTxPoolStore()
+		address1 := types.Address{0x1}
+		testTx1 := newTestTransaction(2, address1)
+		testTx2 := newTestTransaction(4, address1)
+		testTx3 := newTestTransaction(11, address1)
+		address2 := types.Address{0x2}
+		testTx4 := newTestTransaction(7, address2)
+		testTx5 := newTestTransaction(8, address2)
+		mockStore.pending[address1] = []*types.Transaction{testTx1, testTx2}
+		mockStore.pending[address2] = []*types.Transaction{testTx4}
+		mockStore.queued[address1] = []*types.Transaction{testTx3}
+		mockStore.queued[address2] = []*types.Transaction{testTx5}
+		txPoolEndpoint := &TxPool{mockStore}
+
+		result, _ := txPoolEndpoint.ContentFrom(address2)
+
+		response := result.(ContentAddressResponse)
+
+		assert.True(t, mockStore.includeQueued)
+		assert.Equal(t, 1, len(response.Pending))
```

Comment:
Assert queued length as well. You could even compare the two slices (expected and actual) are those equal.